templates/generation/release_please_config_generation.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+
+set -e
+
+echo "{" > .release-please-manifest.json
+
+echo ""\""google-cloud-gapic-bom"\"": "\""0.0.0"\""," >> .release-please-manifest.json
+
+GENERATION_DIR=$(dirname -- "$0");
+
+num_modules="$(wc -l < ../../repos.txt)"
+
+rp_config_line=""
+rp_manifest_line=""
+
+for bom_directory in $(find . -name 'google-*-bom' | sort); do
```

Comment:
Does this (Release Please config) really need to work based on "-bom" directory? To me, it just need to find versions.txt, regardless of "-bom" directories.

You have the second loop which does not rely on "-bom" directory.

Reply:
This is not just generating release-please config. It is also managing versions as we want them. Sometimes it is according to release-please functioning, and right now it is according to upload to staging directory process. 
Also, there are different kind of modules (some having more modules than others), and hence their versions.txt are built differently. 
I'm writing the logic differently now. Earlier assumption that this loop covered all the modules proved to be incorrect, the main challenge here is making the versions sync.

Reply:
> I'm writing the logic differently now. 

Good. Looking forward. Consider creating a loop by finding all "versions.txt" files.

Reply:
As discussed, this script will loop over the modules, generate release-please config files and sync the versions. 
However, there is one category of modules which might not get updated -> single component modules (like java-dns, java-notification, java-grafeas) with both current and released versions as non-snapshot in versions.txt. 
I can fix it later if it poses issues for us.